pkg/reconciler/certificate/certificate.go

```diff
@@ -116,6 +119,10 @@ func (c *Reconciler) reconcile(ctx context.Context, knCert *v1alpha1.Certificate
 		knCert.Status.MarkNotReady(cmCertReadyCondition.Reason, cmCertReadyCondition.Message)
 		return c.setHTTP01Challenges(knCert, cmCert)
 	case cmCertReadyCondition.Status == cmmeta.ConditionTrue:
+		if cmCertRenewingCondition != nil && cmCertRenewingCondition.Reason == renewReason {
+			knCert.Status.MarkNotReady(cmCertReadyCondition.Reason, cmCertReadyCondition.Message)
+			return c.setHTTP01Challenges(knCert, cmCert)
+		}
```

Comment:
This will transition the `Ready` condition from `True` => `Unknown`. This `Unknown` will propagate up the serving route controller and signal that the cert is not ready and things will downgrade to HTTP.

Since renewing doesn't affect the 'active' certificate  I don't think we should be changing the `Ready` condition.

I wonder if instead we should just introduce a new temporary condition to signal that the certificate is being renewed.

Reply:
Yeah, it would be better to add a new condition for the renewing.